scripts/cddl/generate.js

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env node
+
+const fs = require('fs')
+const path = require('path')
+
+const spec = path.resolve(__dirname, '..', '..', 'index.bs')
+const specContent = fs.readFileSync(spec, 'utf-8')
+
+const preFormattedTexts = [...specContent.match(/<pre .*>([^<]*)<\/pre>/g)]
+    /**
+     * filter for CDDL code
+     */
+    .filter((pre) => pre.split('\n')[0].includes('cddl'))
+    /**
+     * drop HTML tags
+     */
+    .map((pre) => pre.split('\n').slice(1, -1))
+    /**
+     * remove obsolete whitespace
+     */
+    .map((pre) => {
+        const preceedingSpace = pre.reduce((prev, line) => {
+            if (line.length === 0) {
+                return prev
+            }
+    
+            const spaces = line.match(/^\s+/)
+            if (spaces) {
+                return Math.min(prev, spaces[0].length)
+            }
+    
+            return 0
+        }, Infinity)
+        return pre.map((line) => line.slice(preceedingSpace))
+    })
+    /**
+     * drop all cddl snippets without declarations, e.g.
+     * ```
+     * EmptyResult
+     * ```
+     */
+    .filter((pre) => pre.length > 1)
+    /**
+     * join code lines
+     */
+    .map((pre) => pre.join('\n'))
+    .join('\n')
+
+fs.writeFileSync(path.join(process.cwd(), 'index.cddl'), preFormattedTexts)
```

Comment:
So I'm surprised we end up with a single schema, because afaict there ought to be two roots; one that matches commands (the "remote end schema") and one that matches responses, events and errors (the "local end schema").

Reply:
We can split it into two CDDL files but all groups or variables used in either of them also need to be defined in both. I am sure we use certain groups in both schemas, don't we?

Reply:
Yeah, we do but the spec tries to do this explicitly by having some fragments defined for both the remote and local end schemas.

Reply:
currently I only see `pre` tags with `local-cddl` and `remote-cddl`. We could add the content of `pre` tags that only have the class `cddl`, e.g. `<pre class="cddl remote-cddl">` to both files. WDYT?

Reply:
I've used `class="local-cddl remote-cddl cddl"` in e.g. the contexts PR. Happy to do whatever is easiest for you here.